cng/pbkdf2.go

```diff
@@ -40,22 +40,24 @@ func PBKDF2(password, salt []byte, iter, keyLen int, h func() hash.Hash) ([]byte
 	}
 	defer bcrypt.DestroyKey(kh)
 	u16HashID := utf16FromString(hashID)
-	buffers := [...]bcrypt.Buffer{
-		{
-			Type:   bcrypt.KDF_ITERATION_COUNT,
-			Data:   uintptr(unsafe.Pointer(&iter)),
-			Length: 8,
-		},
-		{
+	buffers := make([]bcrypt.Buffer, 2, 3)
+	buffers[0] = bcrypt.Buffer{
+		Type:   bcrypt.KDF_ITERATION_COUNT,
+		Data:   uintptr(unsafe.Pointer(&iter)),
+		Length: 8,
+	}
+	buffers[1] = bcrypt.Buffer{
+		Type:   bcrypt.KDF_HASH_ALGORITHM,
+		Data:   uintptr(unsafe.Pointer(&u16HashID[0])),
+		Length: uint32(len(u16HashID) * 2),
+	}
```

Comment:
Could remove that *tiny* bit of manual indexing and pre-counting--I'm fine with it either way:

```suggestion
	buffers := make([]bcrypt.Buffer, 0, 3)
	buffers = append(buffers,
		bcrypt.Buffer{
			Type:   bcrypt.KDF_ITERATION_COUNT,
			Data:   uintptr(unsafe.Pointer(&iter)),
			Length: 8,
		},
		bcrypt.Buffer{
			Type:   bcrypt.KDF_HASH_ALGORITHM,
			Data:   uintptr(unsafe.Pointer(&u16HashID[0])),
			Length: uint32(len(u16HashID) * 2),
		})
```